src/MSBuildLocator/VisualStudioInstanceQueryOptions.cs

```diff
@@ -13,12 +15,23 @@ public class VisualStudioInstanceQueryOptions
         /// </summary>
         public static VisualStudioInstanceQueryOptions Default => new VisualStudioInstanceQueryOptions
         {
-            DiscoveryTypes = DiscoveryType.DeveloperConsole & DiscoveryType.VisualStudioSetup
+            DiscoveryTypes =
+#if FEATURE_VISUALSTUDIOSETUP
+                DiscoveryType.DeveloperConsole | DiscoveryType.VisualStudioSetup
+#endif
+#if NETCOREAPP2_0
+                DiscoveryType.DotNetSdk
+#endif
         };
 
         /// <summary>
         ///     Discovery types for instances included in the query.
         /// </summary>
         public DiscoveryType DiscoveryTypes { get; set; }
+
+        /// <summary>
+        ///     A working directory to use when querying for instances.
```

Comment:
Yeah, this one is pretty awkward: because `global.json` can cause different SDKs to load, you may have to tear down your process to build multiple projects. The only way around that I can think of is having a different AssemblyLoadContext for each, and I don't understand the interop problems that might result. I think we need to make sure we're clear on this one.

```suggestion
        ///     Working directory to use when querying for instances. Ensure it is the project directory to pick up the right global.json.
```